worker/src/test/java/gov/cms/ab2d/worker/service/JobProcessingServiceImplTest.java

```diff
@@ -0,0 +1,199 @@
+package gov.cms.ab2d.worker.service;
+
+import ca.uhn.fhir.context.FhirContext;
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.Job;
+import gov.cms.ab2d.common.model.JobStatus;
+import gov.cms.ab2d.common.model.Sponsor;
+import gov.cms.ab2d.common.model.User;
+import gov.cms.ab2d.common.repository.JobOutputRepository;
+import gov.cms.ab2d.common.repository.JobRepository;
+import gov.cms.ab2d.worker.adapter.bluebutton.BeneficiaryAdapter;
+import gov.cms.ab2d.worker.adapter.bluebutton.BfdClientAdapter;
+import gov.cms.ab2d.worker.adapter.bluebutton.GetPatientsByContractResponse;
+import gov.cms.ab2d.worker.adapter.bluebutton.GetPatientsByContractResponse.PatientDTO;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Resource;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.scheduling.annotation.AsyncResult;
+
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.Future;
+
+import static java.lang.Boolean.TRUE;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+class JobProcessingServiceImplTest {
+    private Random random = new Random();
+
+    @Autowired
+    private FhirContext fhirContext;
+
+    @Mock FileService fileService;
+    @Mock JobRepository jobRepository;
+    @Mock JobOutputRepository jobOutputRepository;
+    @Mock BeneficiaryAdapter beneficiaryAdapter;
+    @Mock BfdClientAdapter bfdClientAdapter;
```

Comment:
A complete _integration_ test will be extremely valuable here. We can avoid mocking all these relatively easily:

- `FileService` does not need to be mocked. It can write to the file system during test.
- You can use the "real" BFD client as well. Take a look at the tests in the `bfd` module. It stands up a mock HTTP server and serves test data (which you can define in XML) in response to FHIR Client REST calls. We should do the same here.

Reply:
Changed the unit test into an integration test that does not mock any of the dependencies.  The integration test will take longer to run than the unit test.